cadcutils/cadcutils/net/ws.py

```diff
@@ -163,24 +162,6 @@ def __init__(self, resource_id, agent, anon=True, cert_file=None, retry=True, ho
         # Unless the caller specifically requests an anonymous client,
         # check first for a certificate, then an externally created
         # HTTPBasicAuth object, and finally a name+password in .netrc.
-        # TODO this should be read from the capabilities of the service
-        if not anon:
-            if (cert_file is not None) and (cert_file is not ''):
-                if os.path.isfile(cert_file):
-                    self.certificate_file_location = cert_file
-                else:
-                    logging.warn("Unable to open supplied certfile ({}). Ignoring."
-                                 .format(cert_file))
-            else:
-                self.basic_auth = auth.get_user_password(self.host)
-        else:
-            self.anon = True
-
-        self.logger.debug(
-            "Client anonymous: {}, certfile: {}, name: {}".format(
-                str(self.anon), str(self.certificate_file_location),
-                str((self.basic_auth is not None) and (self.basic_auth[0]))))
-
         # TODO The service URL needs to be discoverable based on the URI/URL of the service with the
```

Comment:
Agree that the way the service URLs are being built here are pretty ugly and likely should be abstracted and have some reliance on a lookup process.